### PR TITLE
ci: Generalize continuous integration adding support for appveyor and travis

### PR DESCRIPTION
**Pull request by jcfr**
ci: Generalize continuous integration adding support for appveyor and travis

This commit integrates `scikit-ci` and `scikit-ci-addons` to more easily
integrates with continuous integration services.

For more details:
* http://scikit-ci.readthedocs.io
* http://scikit-ci-addons.readthedocs.io

See #101

### CODE REVIEW

**Comment by jcfr** [comment]
@JoostJM Please, to fully test this PR consider enabling Appveyor and Travis.

**Comment by jcfr** [comment]
On travis, the following error is reported:

```
Fatal Python error: PyThreadState_Get: no current thread

/var/folders/my/m6ynh3bn6tq06h7xr3js0z7r0000gn/T/tmpQWv3qu.sh: line 2:  3288 Abort trap: 6           nosetests --with-xunit --logging-level=DEBUG --verbosity=3 tests/test_features.py

A command failed while executing TEST step.

  Return code:

    134

  Command:

    nosetests --with-xunit --logging-level=DEBUG --verbosity=3 tests/test_features.py

  Working directory:

    /Users/travis/build/Radiomics/pyradiomics

Please see above for more information.
```

This is due to the current SimpleITK wheels being linked against Python library. Up-coming release should address this.

**Comment by jcfr** [comment]
@JoostJM  As soon as there is a new wheel of SimpleITK, this PR will be ready. Currently working with @blowekamp to make that happen. (Issue is tracked by https://issues.itk.org/jira/browse/SIMPLEITK-683)

**Comment by blowekamp** [comment]
Can you please verify you are using the 0.10.0 wheel of OS X? Getting pip to download directly from source forge has been very problematic over time due to changing implementation of pip flags. You may just want to try a direct link:
  

> pip install https://sourceforge.net/projects/simpleitk/files/SimpleITK/0.10.0/Python/SimpleITK-0.10.0-cp27-cp27m-macosx_10_6_intel.whl

The 0.10 OS X wheels should not be directly linked to the system python, which is a typical cause of this error you are seeing. Getting the wheel on Pypi would help this issue. I am not sure why a now wheel package would.

**Comment by blowekamp** [comment]
Also this works for me:
`pip install --trusted-host itk.org -f https://itk.org/SimpleITKDoxygen/html/PyDownloadPage.html SimpleITK`

This the actual[ download link URL ](https://github.com/SimpleITK/SimpleITK/blob/master/Wrapping/Python/Packaging/setup.py.in#L68)that easy_install/pip use to follow from pipy.

File: scikit-ci.yml [hunk 1]
```diff
@@ -30,7 +30,8 @@ install:
     - $<RUN_ENV> pip install wheel>=0.29.0
     - $<RUN_ENV> pip install setuptools>=28.0.0
     - $<RUN_ENV> pip install numpy==1.11.0
-    - $<RUN_ENV> pip install --trusted-host www.simpleitk.org -f https://sourceforge.net/projects/simpleitk/files/SimpleITK/0.10.0/Python/ -f http://www.simpleitk.org/SimpleITK/resources/software.html SimpleITK>=0.9.1
+    - $<RUN_ENV> pip install --trusted-host www.simpleitk.org -f https://itk.org/SimpleITKDoxygen/html/PyDownloadPage.html SimpleITK>=0.9.1
```
Comment (blowekamp) [inline scikit-ci.yml]: I believe the trusted host needs to be updated to "itk.org".

Comment (jcfr) [inline scikit-ci.yml]: Good catch. Thanks for the review :+1:

**Comment by jcfr** [comment]
Topic has been rebased. I am now expected it to be green :)

File: scikit-ci.yml [hunk 2]
```diff
@@ -30,7 +30,8 @@ install:
     - $<RUN_ENV> pip install wheel>=0.29.0
     - $<RUN_ENV> pip install setuptools>=28.0.0
     - $<RUN_ENV> pip install numpy==1.11.0
-    - $<RUN_ENV> pip install --trusted-host www.simpleitk.org -f https://sourceforge.net/projects/simpleitk/files/SimpleITK/0.10.0/Python/ -f http://www.simpleitk.org/SimpleITK/resources/software.html SimpleITK>=0.9.1
+    - $<RUN_ENV> pip install --trusted-host www.itk.org -f https://itk.org/SimpleITKDoxygen/html/PyDownloadPage.html SimpleITK>=0.9.1
```
Comment (blowekamp) [inline scikit-ci.yml]: I wonder if www.itk.org vs itk.org will make a difference.

**Comment by jcfr** [comment]
Et voila. This will address #101, #110 and allow to easily enabled testing for #188 

@pieper  @fedorov @JoostJM I plan on merging later this afternoon.

**Comment by fedorov** [comment]
I will need to readthedocs on scikit-ci to understand how this is set up, but otherwise looks great!

**Comment by jcfr** [comment]
Great. Thanks for the review.

For reference, scikit-ci docs is http://scikit-ci.readthedocs.io/en/latest/

**Comment by fedorov** [comment]
Yes, I found it. Looks very helpful!

**Comment by blowekamp** [comment]
Just to follow up. I just uploaded a OS X wheel named `SimpleITK-0.10.0.post3-cp27-cp27m-macosx_10_6_x86_64.whl` to PyPi. 

Now "pip install SimpleITK" works for OS X Python 2.7 too.

Enjoy!

**Comment by jcfr** [comment]
Thanks for the update. That is great progress.

Looking forward, the windows wheels for python 27, and the macosx wheels for 34, 35 and 36

**Pull request merged**